### PR TITLE
[Fluent][Ready for code review] Implemented C# for batch.

### PR DESCRIPTION
**Pull request by anudeepsharma**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anudeepsharma__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ans). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by hovsepm** [comment]
@anudeepsharma you should have AssemblyInfo.cs file for product package.

**Comment by hovsepm** [comment]
@azuresdkci retest this please

File: src/ResourceManagement/Azure/Azure.Tests/BatchTests.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿using Microsoft.Azure.Management.Batch.Models;
```
Comment (hovsepm) [inline src/ResourceManagement/Azure/Azure.Tests/BatchTests.cs]: @anudeepsharma please add license header to all the files.

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            return new ChildListFlattener<IResourceGroup, IBatchAccount>(MyManager.ResourceManager.ResourceGroups.List(), (IResourceGroup resourceGroup) =>
+            {
+                return ListByGroup(resourceGroup.Name);
+            }).Flatten();
+        }
+
+        public PagedList<IBatchAccount> ListByGroup(string resourceGroupName)
+        {
+            var pagedList = new PagedList<AccountResourceInner>(InnerCollection.ListByResourceGroup(resourceGroupName));
+            return WrapList(pagedList);
```
Comment (anuchandy) [inline src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs]: Looks like SDK has `ListByResourceGroup` and `ListByResourceGroupNext`, that means above code will wrap only the first page .. to use the next link, code should be:

```C#
            IPage<AccountResourceInner> firstPage = InnerCollection.ListByResourceGroup(resourceGroupName);
            var pagedList = new PagedList<AccountResourceInner>(firstPage, (string nextPageLink) =>
            {
                return InnerCollection.ListByResourceGroupNext(nextPageLink);
            });
            return WrapList(pagedList);
```

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                MyManager, storageManager);
+        }
+
+        public PagedList<IBatchAccount> List()
+        {
+            return new ChildListFlattener<IResourceGroup, IBatchAccount>(MyManager.ResourceManager.ResourceGroups.List(), (IResourceGroup resourceGroup) =>
+            {
+                return ListByGroup(resourceGroup.Name);
+            }).Flatten();
+        }
```
Comment (anuchandy) [inline src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs]: Batch SDK seems already supports listing across subscription through `List` and `ListNext` so could use that as:

```C#
            IPage<AccountResourceInner> firstPage = InnerCollection.List(resourceGroupName);
            var pagedList = new PagedList<AccountResourceInner>(firstPage, (string nextPageLink) =>
            {
                return InnerCollection.ListNext(nextPageLink);
            });
            return WrapList(pagedList);
```

`ChildListFlattener` is useful when there is no SDK method to list across subscription, in that case its fine to make another call to list resource group.

File: src/ResourceManagement/Azure/Azure.Tests/BatchTests.cs (hunk 1 above)
Comment (anudeepsharma) [inline src/ResourceManagement/Azure/Azure.Tests/BatchTests.cs]: Done

File: src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs (hunk 3 above)
Comment (anudeepsharma) [inline src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs]: Done

File: src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs (hunk 2 above)
Comment (anudeepsharma) [inline src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/BatchAccountsImpl.cs]: Done

File: src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// associated with an assembly.
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("Microsoft.Azure.Management.V2.Batch")]
+[assembly: AssemblyTrademark("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
```
Comment (hovsepm) [inline src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/Properties/AssemblyInfo.cs]: please add 
[assembly: AssemblyVersion("0.0.1.0")]
[assembly: AssemblyFileVersion("0.0.1.0")]

**Comment by azurecla** [comment]
Hi __@anudeepsharma__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ans). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/Batch/global.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "projects": [ "Microsoft.Azure.Management.V2.Batch", "../Resource", "../Storage" ]
```
Comment (hovsepm) [inline src/ResourceManagement/Batch/global.json]: test comment

File: src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (anudeepsharma) [inline src/ResourceManagement/Batch/Microsoft.Azure.Management.V2.Batch/Properties/AssemblyInfo.cs]: Done

**Comment by hovsepm** [comment]
successful CI build log http://azuresdkci.cloudapp.net/view/2-.NET%20SDK/job/sdk-for-net/3343/console

**Pull request merged**